source/include/azure_iot_hub_client.h

```diff
@@ -0,0 +1,411 @@
+ * @param[in] xHmacFunction The #AzureIoTGetHMACFunc_t function pointer to a function which computes the HMAC256 over a set of bytes.
+ * @return An #AzureIoTHubClientResult_t with the result of the operation.
+ */
+AzureIoTHubClientResult_t AzureIoTHubClient_SetSymmetricKey( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle,
+                                                             const uint8_t * pucSymmetricKey,
+                                                             uint32_t ulSymmetricKeyLength,
+                                                             AzureIoTGetHMACFunc_t xHmacFunction );
+
+/**
+ * @brief Connect via MQTT to the IoT Hub endpoint.
+ *
+ * @param[in] xAzureIoTHubClientHandle The #AzureIoTHubClientHandle_t to use for this call.
+ * @param[in] cleanSession A boolean dictating whether to connect with a clean session or not.
+ * @param[in] ulTimeoutMilliseconds The maximum time in milliseconds to wait for a CONNACK.
+ * @return An #AzureIoTHubClientResult_t with the result of the operation.
+ */
+AzureIoTHubClientResult_t AzureIoTHubClient_Connect( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle,
+                                                     bool cleanSession,
+                                                     uint32_t ulTimeoutMilliseconds );
+
+/**
+ * @brief Disconnect from the IoT Hub endpoint.
+ *
+ * @param[in] xAzureIoTHubClientHandle The #AzureIoTHubClientHandle_t to use for this call.
+ * @return An #AzureIoTHubClientResult_t with the result of the operation.
+ */
+AzureIoTHubClientResult_t AzureIoTHubClient_Disconnect( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle );
+
+/**
+ * @brief Send telemetry data to IoT Hub.
+ *
+ * @param[in] xAzureIoTHubClientHandle The #AzureIoTHubClientHandle_t to use for this call.
+ * @param[in] pucTelemetryData The pointer to the buffer of telemetry data.
+ * @param[in] ulTelemetryDataLength The length of the buffer to send as telemetry.
+ * @param[in] pxProperties The property bag to send with the message.
+ * @return An #AzureIoTHubClientResult_t with the result of the operation.
+ */
+AzureIoTHubClientResult_t AzureIoTHubClient_TelemetrySend( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle,
+                                                           const uint8_t * pucTelemetryData,
+                                                           uint32_t ulTelemetryDataLength,
+                                                           AzureIoTMessageProperties_t * pxProperties );
+
+/**
+ * @brief Receive any incoming MQTT messages from and manage the MQTT connection to IoT Hub.
+ *
+ * @note This API will receive any messages sent to the device and manage the connection such as send
+ * `PING` messages.
+ *
+ * @param[in] xAzureIoTHubClientHandle The #AzureIoTHubClientHandle_t to use for this call.
+ * @param[in] ulTimeoutMilliseconds Minimum time (in milliseconds) for the loop to run. If `0` is passed, it will only run once.
```

Comment:
This is how the coreMQTT is set up as well. Do users need to know? Given this timeout, if someone wanted to control how many times it is run, they can throw `processLoop(handle, 0)` in a for loop and let it run.

Reply:
I still don't understand the "only run once" comment.  We might want to discuss at our next sync?  I'm too lazy to look at coreMQTT at the moment but wouldn't 0 just be "run with no timeout, so if there's data waiting to be read return but otherwise don't block"?  I think where I'm not 100% following is that 0 seems like a timeout to me (or lack of one) versus "it will run only once" doesn't necessaryl convey how long this run would take. ?

Reply:
gotcha yea we can discuss

Reply:
OMG John :)